packages/pwa-buildpack/lib/WebpackTools/loaders/wrap-esm-loader.js

```diff
@@ -0,0 +1,112 @@
+const Template = require('webpack/lib/Template');
+
+function wrapEsmLoader(content) {
+    let ids = 0;
+
+    const uniqueJsId = name =>
+        `${Template.toIdentifier(
+            name.includes('/') ? name.slice(name.lastIndexOf('/')) : name
+        )}_${++ids}`;
+
+    const defaultExportRE = /^\s*export\s+default/m;
+    // todo make this actually spec with a parser
+    const exportRE = name =>
```

Comment:
How confident are you in this Regex pattern?

Reply:
Confident enough that it will be easy to match. There will be certain obtuse export styles that defeat it, but that's also true of any Babel plugin.

It would probably be more robust to make this a Babel plugin, but I needed to show an example of using the `source` transform type.